### ISSUE TITLE
OSX Explicit dependency on Ruby 2.0.0

### ISSUE THREAD

**Issue by ReaperUnreal** [etternagame/etterna#81]
I'm getting a crash on startup with the latest Etterna. It's searching for Ruby 2.0.0 explicitly. This is something that was present on OSX 10.11 and 10.12, but 10.13 now has Ruby 2.3.0 by default as seen here:

`ls /System/Library/Frameworks/Ruby.framework/Versions/
2.3/     Current/ `

It would be great if the dependency could be changed to just use Current instead of 2.0. I'm currently installing 2.0 as a workaround but this will definitely need to be fixed at some point to support 10.13.

Included is the full error report.

```
Process:               StepMania [70698]
Path:                  /private/var/folders/*/StepMania.app/Contents/MacOS/StepMania
Identifier:            com.Etterna
Version:               0.55.2 (0.55.2)
Code Type:             X86 (Native)
Parent Process:        ??? [1]
Responsible:           StepMania [70698]
User ID:               1364115093

Date/Time:             2017-10-06 12:43:04.337 -0400
OS Version:            Mac OS X 10.13 (17A365)
Report Version:        12
Anonymous UUID:        34722D3F-2586-9C64-376F-0DC512AC1839

Sleep/Wake UUID:       921A86B8-74BD-4EBC-96FE-34C7501F530A

Time Awake Since Boot: 140000 seconds
Time Since Wake:       2600 seconds

System Integrity Protection: enabled

Notes:                 Translocated Process

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/libruby.2.0.0.dylib
  Referenced from: /private/var/folders/*/StepMania.app/Contents/MacOS/StepMania
  Reason: image not found

Binary Images:
    0x1000 -  0x1257fff +com.Etterna (0.55.2 - 0.55.2) <576EBBF7-72D8-3F23-A41C-0E680FA7E67F> /var/folders/*/StepMania.app/Contents/MacOS/StepMania
 0x1f75000 -  0x1fbafcf  dyld (519.2.1) <3FEC6B17-882D-32AA-9D24-CBD9B1FD18EA> /usr/lib/dyld
0x90015000 - 0x9001cfff  com.apple.agl (3.3.1 - AGL-3.3.1) <53A3D861-48BE-3419-837B-348FA06F2D67> /System/Library/Frameworks/AGL.framework/Versions/A/AGL
0x9001d000 - 0x901d3ff3  com.apple.avfoundation (2.0 - 1334) <8D7020E4-09A5-3E52-AA0E-D6607F336791> /System/Library/Frameworks/AVFoundation.framework/Versions/A/AVFoundation
0x90290000 - 0x90290fff  com.apple.Accelerate (1.11 - Accelerate 1.11) <65375A38-816E-3360-ADF5-F1D466EAE43B> /System/Library/Frameworks/Accelerate.framework/Versions/A/Accelerate
0x9112c000 - 0x91319ff3  com.apple.AddressBook.framework (11.0 - 1805) <11A53C5C-15BE-362A-8A3E-245230D8F5C9> /System/Library/Frameworks/AddressBook.framework/Versions/A/AddressBook
0x9131a000 - 0x920d9ff7  com.apple.AppKit (6.9 - 1561) <F77CDCEC-21B8-36FE-8EEB-12EC5B5F00EC> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
0x920db000 - 0x920dbfff  com.apple.AppKitScripting (6.1 - 32) <476ADFAD-8860-388F-A419-9AD9066E7D91> /System/Library/Frameworks/AppKitScripting.framework/Versions/A/AppKitScripting
0x920dc000 - 0x92119fff  com.apple.AppleScriptKit (1.5.3 - 85) <F83AA7DE-E36A-3157-A29E-8B115BCBCAA7> /System/Library/Frameworks/AppleScriptKit.framework/Versions/A/AppleScriptKit
0x9211a000 - 0x9212afff  com.apple.AppleScriptObjC (1.4 - 33) <15737CB1-0257-3C31-B906-87C9D64C8195> /System/Library/Frameworks/AppleScriptObjC.framework/Versions/A/AppleScriptObjC
0x9212b000 - 0x9212bfff  com.apple.ApplicationServices (48 - 50) <7557388A-2C5B-3467-8793-D56E4B57A1FF> /System/Library/Frameworks/ApplicationServices.framework/Versions/A/ApplicationServices
0x92608000 - 0x92852ffb  com.apple.audio.toolbox.AudioToolbox (1.14 - 1.14) <3D8460F9-6192-3983-83F8-7D2AE16D2DA1> /System/Library/Frameworks/AudioToolbox.framework/Versions/A/AudioToolbox
0x92854000 - 0x92854fff  com.apple.audio.units.AudioUnit (1.14 - 1.14) <146E2F6D-A64B-3AC4-99C0-857700540696> /System/Library/Frameworks/AudioUnit.framework/Versions/A/AudioUnit
0x92855000 - 0x9298affb  com.apple.AudioVideoBridging (600.7 - 600.7) <C86D2784-B3E1-3697-945B-EFBB32171EF9> /System/Library/Frameworks/AudioVideoBridging.framework/Versions/A/AudioVideoBridging
0x9298b000 - 0x92b72fff  com.apple.AutomatorFramework (2.8 - 444.1) <256E1553-E0F8-3C00-B227-98B561B4A253> /System/Library/Frameworks/Automator.framework/Versions/A/Automator
0x92b73000 - 0x92eb2ff7  com.apple.CFNetwork (887 - 887) <379FD334-1AF6-3FA4-8B7C-890995AEC4F0> /System/Library/Frameworks/CFNetwork.framework/Versions/A/CFNetwork
0x92eb4000 - 0x92ec6ff7  com.apple.CalendarStore (8.0 - 1479) <ABE8AA1A-E3FE-3946-AF37-66E351F2C320> /System/Library/Frameworks/CalendarStore.framework/Versions/A/CalendarStore
0x92ec7000 - 0x92ec7fff  com.apple.Carbon (158 - 158) <83C4D410-E216-3EAC-A663-94645F4BCB86> /System/Library/Frameworks/Carbon.framework/Versions/A/Carbon
0x933ce000 - 0x933cefff  com.apple.Cocoa (6.11 - 22) <8DD7C94D-EB2A-30B8-8C39-A446A38AD9DD> /System/Library/Frameworks/Cocoa.framework/Versions/A/Cocoa
0x933cf000 - 0x933dafff  com.apple.Collaboration (80 - 80) <2217DD8C-7D3D-33D3-AE50-E3E6B2B0FD21> /System/Library/Frameworks/Collaboration.framework/Versions/A/Collaboration
0x9349b000 - 0x93536fff  com.apple.audio.CoreAudio (4.3.0 - 4.3.0) <9D1B217E-03F6-38F6-B737-DE5D5DC6DA15> /System/Library/Frameworks/CoreAudio.framework/Versions/A/CoreAudio
0x93538000 - 0x93593ffb  com.apple.audio.CoreAudioKit (1.6.6 - 1.6.6) <CB4FCD5D-1D2E-3ED2-B4E5-01D89DF27EF8> /System/Library/Frameworks/CoreAudioKit.framework/Versions/A/CoreAudioKit
0x93594000 - 0x93599fff  com.apple.CoreBluetooth (1.0 - 1) <E0F40A70-9E04-31FE-A4C8-BE05F6432BAD> /System/Library/Frameworks/CoreBluetooth.framework/Versions/A/CoreBluetooth
0x9359a000 - 0x93879fff  com.apple.CoreData (120 - 847.1) <2AC013DC-8148-34C9-89B1-41353CAECD7F> /System/Library/Frameworks/CoreData.framework/Versions/A/CoreData
0x93881000 - 0x93d09ff7  com.apple.CoreFoundation (6.9 - 1443.13) <31DD4E52-D928-3CF8-BA35-B0080DDBC424> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
0x93d0b000 - 0x94339ffb  com.apple.CoreGraphics (2.0 - 1125.1.1) <4F472247-7953-3DD7-88A4-8DB0065ECC0C> /System/Library/Frameworks/CoreGraphics.framework/Versions/A/CoreGraphics
0x9433b000 - 0x945c0ffb  com.apple.CoreImage (13.0.0 - 573) <6D197299-898E-3BAD-ABB6-444800945B50> /System/Library/Frameworks/CoreImage.framework/Versions/A/CoreImage
0x945c1000 - 0x94631ff7  com.apple.corelocation (2235) <66C77244-E43E-325A-AA52-D594D104ADBF> /System/Library/Frameworks/CoreLocation.framework/Versions/A/CoreLocation
0x94632000 - 0x94678fff  com.apple.audio.midi.CoreMIDI (1.10 - 88) <545850CB-096A-37E3-8FDB-0640B5D222E6> /System/Library/Frameworks/CoreMIDI.framework/Versions/A/CoreMIDI
0x9467b000 - 0x94757ff7  com.apple.CoreMedia (1.0 - 2074.3) <9FBB254B-E142-37EB-98C5-0E851FC5D431> /System/Library/Frameworks/CoreMedia.framework/Versions/A/CoreMedia
0x94758000 - 0x947aafff  com.apple.CoreMediaIO (803.0 - 4976) <A51DE163-E26A-3186-AB73-07B7181D8AF0> /System/Library/Frameworks/CoreMediaIO.framework/Versions/A/CoreMediaIO
0x947ab000 - 0x947abfff  com.apple.CoreServices (822.9 - 822.9) <717212C8-310C-3DDF-8CC9-F0BC46A25225> /System/Library/Frameworks/CoreServices.framework/Versions/A/CoreServices
0x94e3c000 - 0x94f86ff7  com.apple.CoreText (352.0 - 578.3) <8165D2A6-0A1B-3DE7-81B6-F9DCCE480BA6> /System/Library/Frameworks/CoreText.framework/Versions/A/CoreText
0x94f87000 - 0x94fc1ff3  com.apple.CoreVideo (1.8 - 278.0) <D6EF38C6-9ECF-3DDC-BAE5-BA8FFA63A2CB> /System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo
0x94fc2000 - 0x95048ff3  com.apple.framework.CoreWLAN (13.0 - 1335) <54FE3AB9-A361-3D5D-AC9F-9C654AFDECAB> /System/Library/Frameworks/CoreWLAN.framework/Versions/A/CoreWLAN
0x9504a000 - 0x951abff7  com.apple.dvdplayback (5.9.6 - 5960.4.3) <77432E72-362A-34C5-8F6C-BC4AD228C38C> /System/Library/Frameworks/DVDPlayback.framework/Versions/A/DVDPlayback
0x951ac000 - 0x951b6fff  com.apple.DirectoryService.Framework (10.13 - 207) <5F60CAC2-59DD-3457-BF52-F4B8FBF25328> /System/Library/Frameworks/DirectoryService.framework/Versions/A/DirectoryService
0x951b7000 - 0x95276fff  com.apple.DiscRecording (9.0.3 - 9030.4.5) <80DF1650-3C57-352F-B0DC-6CBA11B5CAB1> /System/Library/Frameworks/DiscRecording.framework/Versions/A/DiscRecording
0x95277000 - 0x9529bffb  com.apple.DiscRecordingUI (9.0.3 - 9030.4.5) <B6B11024-4E23-3158-8378-6009F0056B4F> /System/Library/Frameworks/DiscRecordingUI.framework/Versions/A/DiscRecordingUI
0x9529c000 - 0x952a5ff7  com.apple.DiskArbitration (2.7 - 2.7) <E3552A79-57A4-36AE-8B54-5FE2EB5193DA> /System/Library/Frameworks/DiskArbitration.framework/Versions/A/DiskArbitration
0x952ab000 - 0x952adfff  com.apple.ExceptionHandling (1.5 - 12) <2B358A2E-E6FC-3284-9F84-E04CB1F1AF77> /System/Library/Frameworks/ExceptionHandling.framework/Versions/A/ExceptionHandling
0x952ae000 - 0x952b2ffb  com.apple.FWAUserLib (270.10 - 270.10) <FBCA7600-D0A5-38D9-99D9-A7FD9D55004E> /System/Library/Frameworks/FWAUserLib.framework/Versions/A/FWAUserLib
0x952b3000 - 0x952b5ff7  com.apple.ForceFeedback (1.0.6 - 1.0.6) <27F1F703-68C9-3C24-9446-F7804B538336> /System/Library/Frameworks/ForceFeedback.framework/Versions/A/ForceFeedback
0x952b6000 - 0x95623ff7  com.apple.Foundation (6.9 - 1443.14) <2D74E68D-2034-309B-B8B0-E7975A52BA23> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
0x95625000 - 0x95663ff7  com.apple.glut (3.6.14 - GLUT-3.6.9) <BE0DBB27-564B-369F-9A5D-F061090053B3> /System/Library/Frameworks/GLUT.framework/Versions/A/GLUT
0x95664000 - 0x95693fff  com.apple.GSS (4.0 - 2.0) <09F9AD7C-8FD8-3660-9AD4-C6610689682B> /System/Library/Frameworks/GSS.framework/Versions/A/GSS
0x95694000 - 0x95694fff  com.apple.GameKit (558 - 558) <B4AF5659-973A-325F-B056-F5759A480F40> /System/Library/Frameworks/GameKit.framework/Versions/A/GameKit
0x95695000 - 0x956beff7  com.apple.ICADevices (10.0 - 10.1) <34EB5D3A-2A9A-3967-AB3D-DD3154F5784D> /System/Library/Frameworks/ICADevices.framework/Versions/A/ICADevices
0x956bf000 - 0x957d6ff7  com.apple.Bluetooth (6.0.0 - 6.0.0f7) <1005CBBF-C447-33C8-BC90-0E85A1BC0536> /System/Library/Frameworks/IOBluetooth.framework/Versions/A/IOBluetooth
0x957d7000 - 0x9583dff3  com.apple.BluetoothUI (6.0.0 - 6.0.0f7) <17033AE5-CE60-34C0-928F-061B2D30C0C4> /System/Library/Frameworks/IOBluetoothUI.framework/Versions/A/IOBluetoothUI
0x9583e000 - 0x958deffb  com.apple.framework.IOKit (2.0.2 - 1445.1.3) <7D7DD393-BCA5-3E91-8F97-5968CBFC7D3C> /System/Library/Frameworks/IOKit.framework/Versions/A/IOKit
0x958e0000 - 0x958e6fff  com.apple.IOSurface (209 - 209) <CEB65557-BBAB-3233-ABCB-2AE4104DD876> /System/Library/Frameworks/IOSurface.framework/Versions/A/IOSurface
0x958e7000 - 0x95939ff7  com.apple.ImageCaptureCore (7.0 - 7.0) <BDC7F791-3A64-3608-8BF1-0C089C58893D> /System/Library/Frameworks/ImageCaptureCore.framework/Versions/A/ImageCaptureCore
0x9593a000 - 0x95ab7fff  com.apple.ImageIO.framework (3.3.0 - 1694) <58E20069-1F71-3D8C-98EC-7B68BDCFDECD> /System/Library/Frameworks/ImageIO.framework/Versions/A/ImageIO
0x95f27000 - 0x95fceffb  com.apple.InputMethodKit (1.2 - 1.1) <5D321D75-63C8-32DC-986B-E8BCACAC3907> /System/Library/Frameworks/InputMethodKit.framework/Versions/A/InputMethodKit
0x95fcf000 - 0x95fd1fff  com.apple.iChat.InstantMessage (8.0 - 5501) <FEBACD04-5B1D-3CA2-BA5E-7222992B32C2> /System/Library/Frameworks/InstantMessage.framework/Versions/A/InstantMessage
0x95fd4000 - 0x96802ff7  com.apple.JavaScriptCore (13604 - 13604.1.38.1.6) <9D384BD6-E3B9-3D40-8FD3-D72AC358FB8F> /System/Library/Frameworks/JavaScriptCore.framework/Versions/A/JavaScriptCore
0x9681a000 - 0x96832fff  com.apple.Kerberos (3.0 - 1) <8A399DB7-5440-3EC0-A241-3DD10E82DDB2> /System/Library/Frameworks/Kerberos.framework/Versions/A/Kerberos
0x96833000 - 0x96866ffb  com.apple.LDAPFramework (2.4.28 - 194.5) <7A129625-1C7A-34A3-AD2F-0D55D33BE623> /System/Library/Frameworks/LDAP.framework/Versions/A/LDAP
0x96867000 - 0x96889ff7  com.apple.speech.LatentSemanticMappingFramework (2.12.3 - 2.12.3) <2B8EC90C-E74B-39F6-8134-9F1382968906> /System/Library/Frameworks/LatentSemanticMapping.framework/Versions/A/LatentSemanticMapping
0x9688a000 - 0x96892fff  com.apple.MediaAccessibility (1.0 - 114) <087BD9A6-6E5E-3884-B4F7-8F0A6B662B8D> /System/Library/Frameworks/MediaAccessibility.framework/Versions/A/MediaAccessibility
0x96893000 - 0x96e88ffb  com.apple.MediaToolbox (1.0 - 2074.3) <D9CFE9AF-DFC9-3175-BBE4-5A9CA618176F> /System/Library/Frameworks/MediaToolbox.framework/Versions/A/MediaToolbox
0x96e8a000 - 0x96f00ff7  com.apple.Metal (123.3.5 - 123.3.5) <9F8F2C16-236A-39CA-A11F-8D762DB4E37C> /System/Library/Frameworks/Metal.framework/Versions/A/Metal
0x96f02000 - 0x96f0efff  com.apple.NetFS (6.0 - 4.0) <F37A4DA0-AAB6-3F0B-BA18-E322BFA52CC4> /System/Library/Frameworks/NetFS.framework/Versions/A/NetFS
0x96f0f000 - 0x96f47ffb  com.apple.OSAKit (1.5 - 106) <F1C7E956-8CCD-341D-92BC-AAAB7F392468> /System/Library/Frameworks/OSAKit.framework/Versions/A/OSAKit
0x96f48000 - 0x96f6fff7  com.apple.audio.OpenAL (1.8 - 1.8) <E79EEF68-9CE2-3057-B8B7-7B68CF7778EC> /System/Library/Frameworks/OpenAL.framework/Versions/A/OpenAL
0x99c4b000 - 0x99c96fff  com.apple.opencl (2.8.11 - 2.8.11) <A91D3A7A-EACD-3F44-897C-C6DB723CEC03> /System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL
0x99cb4000 - 0x99cbffff  com.apple.OpenDirectory (10.13 - 207) <F5BED4FF-26F2-3580-93ED-4F6B27D99E2F> /System/Library/Frameworks/OpenDirectory.framework/Versions/A/OpenDirectory
0x9ba88000 - 0x9ba96fff  com.apple.opengl (16.0.3 - 16.0.3) <CCA4EA13-DBF6-314D-A6AE-0C273393B425> /System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL
0x9bc50000 - 0x9bc59ff3  com.apple.pcsc (8.0 - 1) <1B9B76F0-9528-3D2F-BEC9-1FDD9D9CC1C9> /System/Library/Frameworks/PCSC.framework/Versions/A/PCSC
0x9bc5a000 - 0x9bc83fff  com.apple.frameworks.preferencepanes (16.0 - 16.0) <814B0995-8A19-33EF-9040-3EEFA618358A> /System/Library/Frameworks/PreferencePanes.framework/Versions/A/PreferencePanes
0x9bc84000 - 0x9bd6cffb  org.python.python (2.7.10 - 2.7.10) <71012EA5-A736-3958-B0AA-075C39097952> /System/Library/Frameworks/Python.framework/Versions/2.7/Python
0x9bd6d000 - 0x9bebeffb  com.apple.QTKit (7.7.3 - 3014) <9DB2668A-3F68-32F0-BF51-C7F791FEE8EE> /System/Library/Frameworks/QTKit.framework/Versions/A/QTKit
0x9c804000 - 0x9c804fff  com.apple.quartzframework (1.5 - 21) <AFDC52CF-3DDF-3100-9D60-9DEFAE551CD7> /System/Library/Frameworks/Quartz.framework/Versions/A/Quartz
0x9c805000 - 0x9ca36fff  com.apple.QuartzCore (1.11 - 574.1.4) <22685244-B08A-3BA4-83EB-7097E0AAA52B> /System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore
0x9ca37000 - 0x9ca8bffb  com.apple.QuickLookFramework (5.0 - 743.3) <5D9F7B15-4E86-346B-B994-38FED5E1ECB5> /System/Library/Frameworks/QuickLook.framework/Versions/A/QuickLook
0x9ca8c000 - 0x9cce4feb  com.apple.QuickTime (7.7.3 - 3014) <A38CF029-378C-3D67-9359-7DC15452B0E4> /System/Library/Frameworks/QuickTime.framework/Versions/A/QuickTime
0xa4584000 - 0xa4659ff3  com.apple.PubSub (1.0.5 - 65.47) <23C98B02-232E-3648-BBA3-DCA917ECA5D8> /System/Library/PrivateFrameworks/PubSub.framework/Versions/A/PubSub
0xa59af000 - 0xa59bbff7  libbz2.1.0.dylib (38) <77C24A36-BE84-3702-A786-935C597A0A86> /usr/lib/libbz2.1.0.dylib
0xa6131000 - 0xa6221ff3  libiconv.2.dylib (51) <FE6D05A5-18DB-3FD8-A52F-B7BADB232C78> /usr/lib/libiconv.2.dylib

Model: MacBookPro9,1, BootROM MBP91.00D7.B00, 4 processors, Intel Core i7, 2.3 GHz, 16 GB, SMC 2.1f173
Graphics: Intel HD Graphics 4000, Intel HD Graphics 4000, Built-In
Graphics: NVIDIA GeForce GT 650M, NVIDIA GeForce GT 650M, PCIe, 512 MB
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1333 MHz, 0x029E, 0x434D5341384758334D314131333333433920
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1333 MHz, 0x029E, 0x434D5341384758334D314131333333433920
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0xF5), Broadcom BCM43xx 1.0 (7.21.190.13.1a3)
Bluetooth: Version 6.0.0f7, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en1
Serial ATA Device: SanDisk SD6SB1M256G, 256.06 GB
Serial ATA Device: MATSHITADVD-R   UJ-8A8
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: IR Receiver
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: FaceTime HD Camera (Built-in)
USB Device: USB 3.0 Bus
USB Device: USB Laser Mouse
Thunderbolt Bus: MacBook Pro, Apple Inc., 25.1
```

**Comment by martensm** [comment]
Thanks for the report. Can you check if this is also an issue on StepMania? AFAIK none of the core Etterna developers have OSX machines to develop on so it's likely we won't be able to reproduce locally.

**Comment by ReaperUnreal** [comment]
I've checked 5.1.0 and 5.0.12 on OSX and both of them boot up fine. I can try downloading master later and giving that a try.

Unfortunately I don't know a whole lot about native OSX applications so I can't help much but I'm more than willing to give things a try.

**Comment by martensm** [comment]
Alright it sounds like a regression on our part then; i'll investigate more at a later time but this is going to be on the backburner for now.

In your first comment you stated you had the intention of downgrading to Ruby 2.0 as a workaround -- can you confirm that you have done this and it fixed the crash?

**Comment by ReaperUnreal** [comment]
I'm in the process of doing this, it turns out it's far more complicated than it seems, but I'll post here if/when I succeed.

**Comment by ReaperUnreal** [comment]
I've managed to find the 2.0 version of the Ruby framework for OSX and put it in the appropriate place with the appropriate flags. Sadly there's another issue after this where it claims that no noteskins can be found. However there's definitely noteskins in the appropriate folder. I then ran it from the command line to get more debug info and came away with this:

```objc[10663]: Class VCWeakObjectHolder is implemented in both /System/Library/PrivateFrameworks/AVConference.framework/Frameworks/ViceroyTrace.framework/Versions/A/ViceroyTrace (0xa8c0a120) and /System/Library/PrivateFrameworks/AVConference.framework/Versions/A/AVConference (0xa8c252f8). One of the two will be used. Which one is undefined.
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.1.1/Common/ChunkCompression.cpp:50: Error: unsupported compressor 8
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.1.1/Libraries/CompressData/CompressData.c:353: Error: Unknown compression scheme encountered for file '/System/Library/CoreServices/CoreTypes.bundle/Contents/Resources/Exceptions.plist'
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.1.1/Common/ChunkCompression.cpp:50: Error: unsupported compressor 8
/BuildRoot/Library/Caches/com.apple.xbs/Sources/AppleFSCompression/AppleFSCompression-96.1.1/Libraries/CompressData/CompressData.c:353: Error: Unknown compression scheme encountered for file '/System/Library/CoreServices/CoreTypes.bundle/Contents/Library/AppExceptions.bundle/Exceptions.plist'
Etterna0.55-git-c692da02d5
Compiled 20171003 @ 21:03:19 (build c692da02d5)
Log starting 2017-10-09 20:47:51
Loading window: macosx
Model: MacBookPro9,1 (8/8)
Clock speed 2.30 GHz
Mac OS X 10.13
Memory: 16.00 GB```

Which sounds like yet another library that Apple updated without telling anyone. I'm going to attempt to compile my own version and see if that fixes things. If not, maybe I can trace down where the dependencies are all set.

**Comment by martensm** [comment]
Somebody has reported successfully running the 0.55.2 release `.dmg` on OSX 10.12 but they haven't tried 10.13. I suspect that it's an issue with how the `.dmg` is packaged and what it expects for system library versions.

Thanks again for investigating this.

CC @TheROPFather

**Comment by LiamEarle** [comment]
Just confirming updating to 0.55.3 fixes the dependency issues on macOS 10.13 (High Sierra)

**Comment by ReaperUnreal** [comment]
I can confirm that the dependency issues are fixed for me as well with 0.55.3. Thanks!

**Comment by martensm** [comment]
Going to leave this open for now as a reminder to remove other unnecessary dependencies since that did indeed turn out to be the issue.
